### PR TITLE
Fix #7 

### LINKED ISSUE #7: Potential bug
The structure of the train_step in cell 8 of the notebook is very unconventional.

```
def train_step(self, data):
... first model evaluation
... first tape gradient
... second model evaluation
... update parameters
... second tape gradient      
```

Usually for the parameter update to affect the second tape gradient the update shall be before the second model evaluation. 
```
def train_step(self, data):
... first model evaluation
... first tape gradient
... update parameters
... second model evaluation
... second tape gradient      

```

### PR DESCRIPTION
**Pull request by rainwoodman**
Fix #7.

The SAM training accuracy is near 82.5%, increased from the quoted numbers on the home page.

The non-SAM training appears to terminated early at 5m30, ~ 70 epochs, and at a lower accuracy than the home page number.

### CODE REVIEW

**Comment by rainwoodman** [comment]
I am not sure if this ipynb file is clean -- I see some references to my own github repo.

**Comment by sayakpaul** [comment]
@rainwoodman the notebook is great. Just a few minor suggestions:

* In the cell that implements `SAM` class I guess we can skip the TensorFlow import. Also, I don't think we need to disable eager execution for the current setup. My understanding is that when `train_step()` is overridden and the underlying model is compiled it is going to run in graph mode already. However, please correct me if I am missing out on something here. 
* You are welcome to edit the README numbers with the latest ones. This will give the users more confidence in using SAM I hope.

**Comment by rainwoodman** [comment]
You are right I forgot to remove the lines. Those were for debugging, when I checked that the before this change indeed we used to have gradients == sam_gradients.

**Comment by rainwoodman** [comment]
I removed the lines, and updated README. Please take a look, thanks!

File: README.md [hunk 1]
```diff
@@ -27,8 +27,8 @@ SAM has only one hyperparameter namely `rho` that controls the neighborhood of t
 | Without SAM |            0.575114            |           82.9          |
```
Comment (sayakpaul) [inline README.md:27]: Did you also want to update this table with the final scores?

**Review by sayakpaul — CHANGES_REQUESTED** [review]
@rainwoodman thank you for the changes. Just one more minor comment.

File: README.md (hunk 1 above)
Comment (rainwoodman) [reply README.md:27]: Yes It was updated in the previous commit. ;)

Comment (sayakpaul) [reply README.md:27]: Oh I see. Anyway the current README should reflect the changes you made. Thank you so much.
